google-cloud-spanner/acceptance/spanner/client/transaction_test.rb

```diff
@@ -126,6 +128,83 @@
       assert_accounts_equal expected, actual
     end
   end
+  focus
+  it "supports tx isolation with read and update" do
+    results = db.read "accounts", [:reputation], keys: 1, limit: 1
+    original_val = results.rows.first[:reputation]
+    begin
+      latch = Concurrent::CountDownLatch.new 1
+      thr_1 = Thread.new do
+        db.transaction do |tx|
+          tx_results = tx.read "accounts", [:reputation], keys: 1, limit: 1
+          puts "read 1"
+          tx_val = tx_results.rows.first[:reputation]
+          new_val = tx_val + 1
+          tx.update "accounts", [{ account_id: 1, reputation: new_val }]
+          puts "write 1"
+          latch.wait
```

Comment:
I copied the [Java test](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITTransactionTest.java#L108), which appears to me to be waiting after the update.